From 36a2b6dbb9b7839dcb2127bb5d6a5d90c520345f Mon Sep 17 00:00:00 2001
From: Himal Ghimiray <himalg@codeaurora.org>
Date: Fri, 27 Oct 2017 15:37:06 +0530
Subject: meta-qti-bsp: Adding users in LE.

Added system,radio,bluetooth,graphics,input,audio,
camera,adb,sensors,Apps,nobody as users.

Change-Id: I2aaaae932c696cdd84c85b27f3fcdc87244f686a
---
 conf/distro/msm-perf.conf | 27 ++++++++++++++++++++++++++-
 conf/distro/msm-user.conf | 20 ++++++++++++++++++++
 conf/distro/msm.conf      | 27 ++++++++++++++++++++++++++-
 3 files changed, 72 insertions(+), 2 deletions(-)

diff --git a/conf/distro/msm-perf.conf b/conf/distro/msm-perf.conf
index 4b642d7..18d809e 100644
--- a/conf/distro/msm-perf.conf
+++ b/conf/distro/msm-perf.conf
@@ -3,7 +3,7 @@
 #@NAME: MSM-PERF
 #@DESCRIPTION: QTI Linux Distribution for MSM targets with performance orientation (glibc based)
 ################################################################################################
-
+INHERIT += "extrausers"
 require conf/distro/include/msm.inc
 
 # DISTRO CONFIGURATION
@@ -27,3 +27,28 @@ PERF_BUILD = "1"
 # Append variants like '-perf', '-user', 'perf-minimal' etc to deploy dir.
 DEPLOY_NAME ?= "${MACHINE}${@['-' + d.getVar('PRODUCT', True), ''][d.getVar('PRODUCT', True) == ('' or 'base')]}${@['-' + d.getVar('VARIANT', True), ''][d.getVar('VARIANT', True) == ('')]}"
 DEPLOY_DIR_IMAGE = "${DEPLOY_DIR}/images/${DEPLOY_NAME}"
+
+EXTRA_USERS_PARAMS = "\
+ useradd -u 1000 -U system; \
+ useradd -u 1001 -U radio; \
+ useradd -u 1002 -U bluetooth; \
+ useradd -u 1003 -U graphics; \
+ useradd -u 1004 -g input input; \
+ useradd -u 1005 -g audio audio; \
+ useradd -u 1006 -U camera;\
+ useradd -u 2000 -U adb; \
+ useradd -u 3011 -U sensors; \
+ useradd -u 4000 -U apps; \
+ useradd -u 9999 -U nobody; \
+ usermod -L -e 1 system; \
+ usermod -L -e 1 radio; \
+ usermod -L -e 1 bluetooth; \
+ usermod -L -e 1 graphics; \
+ usermod -L -e 1 input; \
+ usermod -L -e 1 audio; \
+ usermod -L -e 1 camera; \
+ usermod -L -e 1 adb; \
+ usermod -L -e 1 sensors; \
+ usermod -L -e 1 Apps; \
+ usermod -L -e 1 nobody; \
+"
diff --git a/conf/distro/msm-user.conf b/conf/distro/msm-user.conf
index ef98d47..8c64971 100644
--- a/conf/distro/msm-user.conf
+++ b/conf/distro/msm-user.conf
@@ -33,5 +33,25 @@ DEPLOY_NAME ?= "${MACHINE}${@['-' + d.getVar('PRODUCT', True), ''][d.getVar('PRO
 DEPLOY_DIR_IMAGE = "${DEPLOY_DIR}/images/${DEPLOY_NAME}"
 
 EXTRA_USERS_PARAMS = "\
+ useradd -u 1000 -U system; \
+ useradd -u 1001 -U radio; \
+ useradd -u 1002 -U bluetooth; \
+ useradd -u 1003 -U graphics; \
+ useradd -u 1006 -U camera;\
+ useradd -u 2000 -U adb; \
+ useradd -u 3011 -U sensors; \
+ useradd -u 4000 -U apps; \
+ useradd -u 9999 -U nobody; \
+ usermod -L -e 1 system; \
+ usermod -L -e 1 radio; \
+ usermod -L -e 1 bluetooth; \
+ usermod -L -e 1 graphics; \
+ usermod -L -e 1 input; \
+ usermod -L -e 1 audio; \
+ usermod -L -e 1 camera; \
+ usermod -L -e 1 adb; \
+ usermod -L -e 1 sensors; \
+ usermod -L -e 1 Apps; \
+ usermod -L -e 1 nobody; \
  usermod -L -e 1 root; \
 "
diff --git a/conf/distro/msm.conf b/conf/distro/msm.conf
index b1972c0..7b128e8 100644
--- a/conf/distro/msm.conf
+++ b/conf/distro/msm.conf
@@ -3,7 +3,7 @@
 #@NAME: MSM
 #@DESCRIPTION: MSM Linux Distribution (glibc based)
 #############################################################################
-
+INHERIT += "extrausers"
 require conf/distro/include/msm.inc
 
 # DISTRO CONFIGURATION
@@ -24,3 +24,28 @@ DISTRO_FEATURES += "emmc-boot"
 # Append variants like '-perf', '-user', 'perf-minimal' etc to deploy dir.
 DEPLOY_NAME ?= "${MACHINE}${@['-' + d.getVar('PRODUCT', True), ''][d.getVar('PRODUCT', True) == ('' or 'base')]}${@['-' + d.getVar('VARIANT', True), ''][d.getVar('VARIANT', True) == ('')]}"
 DEPLOY_DIR_IMAGE = "${DEPLOY_DIR}/images/${DEPLOY_NAME}"
+
+EXTRA_USERS_PARAMS = "\
+ useradd -u 1000 -U system; \
+ useradd -u 1001 -U radio; \
+ useradd -u 1002 -U bluetooth; \
+ useradd -u 1003 -U graphics; \
+ useradd -u 1004 -g input input; \
+ useradd -u 1005 -g audio audio; \
+ useradd -u 1006 -U camera;\
+ useradd -u 2000 -U adb; \
+ useradd -u 3011 -U sensors; \
+ useradd -u 4000 -U apps; \
+ useradd -u 9999 -U nobody; \
+ usermod -L -e 1 system; \
+ usermod -L -e 1 radio; \
+ usermod -L -e 1 bluetooth; \
+ usermod -L -e 1 graphics; \
+ usermod -L -e 1 input; \
+ usermod -L -e 1 audio; \
+ usermod -L -e 1 camera; \
+ usermod -L -e 1 adb; \
+ usermod -L -e 1 sensors; \
+ usermod -L -e 1 Apps; \
+ usermod -L -e 1 nobody; \
+"
-- 
cgit v1.1

